src/Aspire.Hosting.Azure/Utils/ResourceGroupNameHelpers.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using System.Text;
+
+namespace Aspire.Hosting.Azure.Utils;
+
+internal static class ResourceGroupNameHelpers
+{
+    public static int MaxResourceGroupNameLength = 90;
+
+    /// <summary>
+    /// Excludes any characters which are not valid resource group name components.
+    /// </summary>
+    /// <param name="text">The text to normalize.</param>
+    /// <returns>The normalized resource group name or an empty string if no characters were valid.</returns>
+    public static string NormalizeResourceGroupName(string text)
+    {
+        ArgumentNullException.ThrowIfNull(text);
+
+        var stringBuilder = new StringBuilder(capacity: text.Length);
+
+        for (var i = 0; i < text.Length; i++)
+        {
+            var c = text[i];
+
+            // Only be a letter, digit, '-', '.', '(', ')' or '_'
+            if (!char.IsLetterOrDigit(c) && c != '-' && c != '.' && c != '(' && c != ')' && c != '_')
+            {
+                continue;
+            }
+
+            stringBuilder.Append(c);
+        }
+
+        // Can't end with '.'
+        return stringBuilder.ToString().TrimEnd('.');
+    }
+
+    /// <summary>
+    /// Try to convert or excludes any characters which are not valid resource group name components for AZD.
```

Comment:
> for AZD

This is kind of weird. Why are we special casing "for AZD"?

Reply:
@timheuer 's comment was that if we ask AZD to create an RG that it won't accept then this could be an issue.

Reply:
I think it is fine to make this a comment, but naming the method "ForAzd" seems super weird.

Reply:
Removed the mention.